### DISCUSSION TITLE
Can I force yazi to use chafa?

### DISCUSSION THREAD

**Discussion by MarvinJWendt** [sxyazi/yazi#1470] [1. Q&A]
Hi, I want to use chafa, but I cannot get yazi to use it. 
Yazi always detects either Wayland or X11, which is not available on my system (Windows Terminal with Arch in WSL).
Can I somehow force yazi to use the chafa adapter?

**Comment by sxyazi** [comment]
I'm not familiar with WSL - do you mean you're running X11/Wayland in WSL and have the `XDG_SESSION_TYPE` env variable set?

**Reply by MarvinJWendt** [reply]
I do have some hacks in place in my WSL to make some tools work. It simulates a display, which sets the DISPLAY env var. There is no Wayland/X11 running, but some of those env variables are set. Yazi isn't wrong by not detecting that, but I would like to be able to force set an adapter, as the one detected by yazi will not work.

**Reply by sxyazi** [reply]
If that's the case, I think you can start Yazi with `DISPLAY="" yazi`

**Reply by MarvinJWendt** [reply]
![image](https://github.com/user-attachments/assets/e9db5414-11f7-4424-8a23-3bcc7a4218cf)

Still matches Wayland somehow

**Reply by MarvinJWendt** [reply]
I guess this is because of `WAYLAND_DISPLAY`

```
Desktop
    XDG_SESSION_TYPE: None
    WAYLAND_DISPLAY : Some("wayland-0")
    DISPLAY         : Some("")
```

But even if I unset it too, it matches `KittyOld`:
![image](https://github.com/user-attachments/assets/32d9c577-84ee-44b5-a25a-dcd118946cc1)

**Reply by MarvinJWendt** [reply]
Even when I use the `unset` command to "delete" the env var completely from my system, it still detects `KittyOld`:
```
Adapter
    Adapter.matches: KittyOld

Desktop
    XDG_SESSION_TYPE: None
    WAYLAND_DISPLAY : None
    DISPLAY         : None
```

**Reply by sxyazi** [reply]
It looks like Yazi mistook MS Terminal for WezTerm. Would you like to file a bug report for it? I think it should be easy to fix.

**Reply by MarvinJWendt** [reply]
Done, sorry for the delay. (#1611)

**Comment by sxyazi** [comment]
Done in https://github.com/sxyazi/yazi/pull/1735
